silhouette-password-bcrypt/src/main/scala/com/mohiva/play/silhouette/password/BCryptPasswordHasher.scala

```diff
@@ -51,7 +47,11 @@ case class BCryptPasswordHasher(logRounds: Int = 10) extends PasswordHasher {
    * @param plainPassword The password to hash.
    * @return A PasswordInfo containing the hashed password.
    */
-  override def hash(plainPassword: String) = PasswordInfo(id, BCrypt.hashpw(plainPassword, BCrypt.gensalt(logRounds)))
+  override def hash(plainPassword: String) = PasswordInfo(
+    hasher = id,
+    password = BCrypt.hashpw(plainPassword, BCrypt.gensalt(logRounds)),
+    metadata = Some(logRounds.toString)
```

Comment:
You are completely right! This is my [first comment of your initial issue](https://github.com/mohiva/play-silhouette/pull/454#issuecomment-201814287):

> A bcrypt password hash includes the rounds the hash wars created with. So there is no need to save the number of rounds otherwise.

I've absolutely no idea why I came to the idea to store the log rounds in an extra field. Maybe because of #459.

I'll update the pull request so that the `isDeprecated` method extracts the log rounds from the password hash.